src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,115 @@
+import "@testing-library/jest-dom";
+import fetchMock from "jest-fetch-mock";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import App from "../App";
+
+// Before each test, reset and enable fetch mocking
+beforeEach(() => {
+  fetchMock.resetMocks();
+  fetchMock.enableMocks();
+});
+
+describe("App", () => {
+  it("should form not be in the document initially", () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    expect(screen.queryByRole("input")).not.toBeInTheDocument();
+  });
+
+  it("should open form when CTA button is clicked", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    const newCardInput = await screen.findByRole("textbox");
+    expect(newCardInput).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when front card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
```

Comment:
I can't disable this button because I use the same button to save the card. I render different text and point to a different function when filling the input on the back of the card (this is still the same button). 
The same applies to the input that refers to a specific state of the currently displayed card page. Is this a bad approach? should I render a second form for the back of the card?

```
          <Button
            onClick={isFront ? nextStepHandler : saveCard}
            variant="primary"
          >
            {isFront ? "Next" : "Save"}
          </Button>
```
          
        <Input
          className={styles.input}
          onChange={inputValueHandler}
          value={isFront ? firstStepInputValue : secondStepInputValue}
        />

Reply:
Hm, but you should not see `Save` button when clicking `Next` when first value is falsy. Anyway, you can also pass there `disabled`, but you'd need to make this check more complex, something like: `disabled={isFront ? !frontValue : !backValue}`